scene/resources/default_theme/default_theme.cpp

```diff
@@ -803,9 +803,11 @@ void fill_default_theme(Ref<Theme> &theme, const Ref<Font> &default_font, const
 	theme->set_font_size("font_size", "ItemList", -1);
 
 	theme->set_color("font_color", "ItemList", control_font_lower_color);
+	theme->set_color("font_hovered_color", "ItemList", control_font_hover_color);
 	theme->set_color("font_selected_color", "ItemList", control_font_pressed_color);
 	theme->set_color("font_outline_color", "ItemList", Color(1, 1, 1));
 	theme->set_color("guide_color", "ItemList", Color(0.7, 0.7, 0.7, 0.25));
+	theme->set_stylebox("hovered", "ItemList", make_flat_stylebox(Color(1, 1, 1, 0.07)));
```

Comment:
Default theme should not be changed and should provide values consistent with the current look, so existing projects keep their appearance.

Reply:
We are in a stable phase right now and must preserve compatibility whenever possible. There are no expectations that Godot 3 projects will work and look the same in Godot 4. There is such expectation between 4.0 and 4.1.

There is no reason why this should be enabled by default for everyone when this wasn't the case before. Those who need this for their project will update their theme accordingly.

Reply:
I think people who use the default theme in their project don't expect that it will never change at all. And this isn't some big change, it's just a highlight for list items on mouse hover. Most people will appreciate it or won't mind at all.

Reply:
This affects everyone, not just people who never change their theme from the default. Since this adds new properties, they will not be overridden in user themes.

Reply:
See also https://github.com/godotengine/godot/pull/29268.

I would really prefer this to be enabled by default in projects. Otherwise, it's hardly going to see any use at all.

Reply:
We don't really plan any special compatibility breakage for 4.1 which would not be possible for any future 4.x. The same rules will apply going forward and starting now.

It is indeed a small change, so if you insist, alright then, let's keep it.

Reply:
Fair enough! I get it about the stability requirement between minor versions, and this totally makes sense. Godot gets more traction, game studios are moving to it and we want to avoid unpleasant surprises on their projects after an upgrade.

My vote is to have this change in. But we are a community and should decide together. So, go or no-go for this addition?

I have other open PRs with similar improvements on other types of controls. 
If we do not have a guideline about that acceptable scope of theme "breakage", I suggest we document it so that other contributors can be on the same wavelength.

Other options to avoid the surprise would be a good enough statement in the changelog, or detect that a project is from 4.0 and convert it at the moment it is opened.

Reply:
> So, go or no-go for this addition?

Keep it 🙃